### DISCUSSION TITLE
Budget caps

### DISCUSSION THREAD

**Discussion by MoIocSidi** [microsoft/AzureTRE#3389] [Ideas]
Is it possible to make a workspace inaccessible after a budgetary amount is exceeded?

**Comment by marrobi** [comment]
Hi @MoIocSidi at present no as we do not have the concept of a "budget".

The challenge is the TRE does not managed the underlying Azure services. So for example with Azure ML, or Databricks, it's not as simple as blocking access. However we do have a work item to create specific disable actions for each service - https://github.com/microsoft/AzureTRE/issues/2168 .

Azure cost management has budget management facility with action groups. Between the two we could likely come up with a solution.

Maybe you could create a feature request- https://github.com/microsoft/AzureTRE/issues/new?assignees=&labels=feature&template=feature_request.md&title= which might start a wider discussion.

**Comment by SvenAelterman** [comment]
@MoIocSidi You might also want to take a look at the work US EDU has done to stop/shutdown services that can be shut down. This is combined with a budget alert in Cost Management.

https://github.com/microsoft/STRIDES/blob/main/budgetDeallocator/README.md
